clients/admin-ui/src/features/common/table/v2/column-settings/ColumnSettingsModal.tsx

```diff
@@ -0,0 +1,124 @@
+import {
+  Box,
+  Button,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Tab,
+  TabList,
+  TabPanel,
+  TabPanels,
+  Tabs,
+  Text,
+} from "@fidesui/react";
+import { Table as TableInstance } from "@tanstack/react-table";
+import { useCallback, useMemo } from "react";
+
+import {
+  DraggableColumn,
+  DraggableColumnList,
+  useEditableColumns,
+} from "./DraggableColumnList";
+
+type ColumnSettingsModalProps<T> = {
+  isOpen: boolean;
+  onClose: () => void;
+  prefixColumns: string[];
+  tableInstance: TableInstance<T>;
+};
+
+export const ColumnSettingsModal = <T,>({
+  isOpen,
+  onClose,
+  tableInstance,
+  prefixColumns,
+}: ColumnSettingsModalProps<T>) => {
+  const initialColumns = useMemo(
+    () =>
+      tableInstance
+        .getAllColumns()
+        .filter((c) => !prefixColumns.includes(c.id))
+        .map((c) => ({
+          id: c.id,
+          displayText: c.columnDef?.meta?.displayText || c.id,
+          isVisible: c.getIsVisible(),
+        })),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
+  const columnEditor = useEditableColumns({
+    columns: initialColumns,
+  });
+
+  const handleSave = useCallback(() => {
+    tableInstance.setColumnOrder([
+      ...prefixColumns,
+      ...columnEditor.columns.map((c) => c.id),
+    ]);
+    tableInstance.setColumnVisibility(
+      columnEditor.columns.reduce(
+        (acc: Record<string, boolean>, current: DraggableColumn) => {
+          // eslint-disable-next-line no-param-reassign
+          acc[current.id] = current.isVisible;
+          return acc;
+        },
+        {}
+      )
+    );
+    onClose();
+  }, [onClose, prefixColumns, tableInstance, columnEditor.columns]);
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} isCentered size="2xl">
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader pb={0}>Data Map Settings</ModalHeader>
```

Comment:
We should parameterize the modal header if we plan on this being a reusable component